include/mesh.h

```diff
@@ -333,6 +344,8 @@ class Mesh {
   Container<ParticleBase<Tdim>> particles_;
   //! Container of particle sets
   tsl::robin_map<unsigned, Container<ParticleBase<Tdim>>> particle_sets_;
+  //! Container of remove steps
+  tsl::robin_map<mpm::Index, std::vector<unsigned>> remove_steps_;
```

Comment:
Let's remove this map and use the existing `particle_sets_`

Reply:
I think the mpm::Index is related to the time step, so we should have this in the explicit algorithm rather than the mesh.

Reply:
mesh doesn't have a sense of time

Reply:
Yes, the key is the number of remove step, and the vector is the ids of particle sets which need to be removed. Do you mean store this map in the mpm_explicit straightly?